tests/cpp/helpers.cc

```diff
@@ -357,11 +359,15 @@ std::unique_ptr<DMatrix> CreateSparsePageDMatrix(
     row_count += batch.Size();
   }
 #if defined(_OPENMP)
-  EXPECT_GE(batch_count, 2);
-  EXPECT_EQ(row_count, dmat->Info().num_row_);
+  if (omp_get_max_threads() != 1) {
+    EXPECT_GE(batch_count, 2) << " Threads: " << omp_get_max_threads();
+    EXPECT_EQ(row_count, dmat->Info().num_row_);
+  }
```

Comment:
FYI, this change might help us transition to Ninja build (#5832)

Reply:
The root cause is in external memory.  We need to find a time to put a new impl together.

Reply:
Time to face the monster, then.

Reply:
Shall I open a new issue to keep track of the problem with external memory?